sigma/backends/loki/loki.py

```diff
@@ -1039,12 +1083,25 @@ def finalize_query(
         elif query is None:
             query = ""
         if state.has_deferred():
-            query = self.deferred_separator.join(
-                (
-                    deferred_expression.finalize_expression()
-                    for deferred_expression in state.deferred
-                )
-            ) + (" " + query if len(query) > 0 else "")
+            standard_deferred = [
+                expression.finalize_expression()
+                for expression in state.deferred
+                if not isinstance(expression, LogQLDeferredFieldRefExpression)
+            ]
+            field_refs = [str(self.select_log_parser(rule))] + [

```

Comment:
This is going to inevitably introduce two log parsers if there are any queries that have both `|fieldref`s and normal field-value conditions (which I have a feeling will likely break some queries) - is there a particular reason that the `field_refs` can't be added **after** the query and thereby rely on the first log parser call (**if** there is one, adding one if not)? This is especially relevant as this operation will likely be the slowest part of the query, so filtering out any irrelevant lines before doing it would be best!